src/kOS/Communication/CommNetConnectivityManager.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using CommNet;
+using kOS.Module;
+
+namespace kOS.Communication
+{
+    public class CommNetConnectivityManager : IConnectivityManager
+    {
+        private CommPath tempPath = new CommPath();
+
+        /// <summary>
+        /// Checks to see if CommNet is enabled in the current game.  This should be checked often
+        /// because CommNet itself and kOS's CommNet integration can be enabled and disabled from
+        /// the in game dificulty settings menu. True if all CommNet support is enabled.
+        /// </summary>
+        public bool IsEnabled
+        {
+            get
+            {
+                return HighLogic.CurrentGame != null
+                    && HighLogic.CurrentGame.Parameters.Difficulty.EnableCommNet;
+            }
+        }
+
+        public double GetDelay(Vessel vessel1, Vessel vessel2)
+        {
+            if (!IsEnabled)
+                return 0;
+            if (HasConnection(vessel1, vessel2))
+                return 0; // CommNet.CommPath does not currently expose a delay value
+            return -1;
+        }
+
+        public double GetDelayToHome(Vessel vessel)
+        {
+            if (!IsEnabled)
+                return 0;
+            if (HasConnectionToHome(vessel))
+                return vessel.Connection.SignalDelay; // in stock this is a dummy 0, but other mods may modify it
+            return -1;
+        }
+
+        public double GetDelayToControl(Vessel vessel)
+        {
+            return 0;
+        }
+
+        public bool HasConnectionToHome(Vessel vessel)
+        {
+            if (!IsEnabled)
+                return true;
+
+            // IsConnectedHome is only set to true on the active vessel, so we have to manually
+            // call the FindHome method to evaluate the home connection.
+
+            // WARNING: In stock this will only work for vessels with a relay antenna installed.
+            // This is a limitation put in place to improve performance in the stock game, and
+            // there isn't a very good way around it.
```

Comment:
That is correct.  I have it documented, but I don't do it in a big "note" box, which may be a good decision.  I agree it's silly.  At least I was able to work around it a little for inter-vessel communication by trying the path both ways.

I can update the documentation.